src/components/IncomingMessageList/index.jsx

```diff
@@ -212,17 +212,27 @@ export class IncomingMessageList extends Component {
     this.props.onConversationSelected(rowsSelected, selectedConversations);
   };
 
-  handleOpenConversation = index => {
-    const conversation = this.props.conversations.conversations.conversations[
-      index
-    ];
-    this.setState({ activeConversation: conversation });
+  handleOpenConversation = index =>
+    this.setState({ activeConversationIndex: index });
+
+  handleRequestPreviousConversation = () => {
+    const { activeConversationIndex: oldIndex } = this.state;
+    const newIndex = oldIndex - 1;
+    if (newIndex < 0) return;
+    this.setState({ activeConversationIndex: newIndex });
   };
 
-  handleCloseConversation = () => {
-    this.setState({ activeConversation: undefined });
+  handleRequestNextConversation = () => {
+    const { activeConversationIndex: oldIndex } = this.state;
+    const { conversations } = this.props.conversations.conversations;
+    const newIndex = oldIndex + 1;
+    if (newIndex >= conversations.length) return;
+    this.setState({ activeConversationIndex: newIndex });
```

Comment:
Idiomatic way to do this is:

```suggestion
    this.setState(prevState => {
      const { activeConverastionIndex: oldIndex } = this.state;
      const { conversations } = this.props.conversations.conversations;
      const newIndex = oldIndex + 1;
      if (newIndex >= conversations.length) return;
      return { activeConversationIndex: newIndex };
    });
```

That way if someone spams the button, it'll increment – otherwise, because `this.setState` is async, the next `handleRequestNextConversation` may have the old state